### ISSUE TITLE
Implement a timeline filter based on the source software or source instance of the first message in a conversation

### ISSUE THREAD

**Issue by informapirata** [friendica/friendica#14112]
Good morning everyone,

### Is the feature request related to a problem? Please describe.

This request is not related to an issue, but is about requesting additional ways to filter users' timelines.
In fact, Friendica has the possibility of filtering the timelines based on the origin protocol of the first message of a conversation (see pic).

![image](https://github.com/friendica/friendica/assets/99956935/7527c99e-9c35-4d45-8aae-b08e838a8ee9)

### Describe the feature you'd like

However, since ActivityPub is now the protocol used by 98% of applications, it would be interesting to be able to have the same type of filter:

1) based on the source software: Lemmy/Kbin, Wordpress/Writefreely, Peertube, Funkwhale are in fact software that characterize messages in a different way compared to classic microblogging
2) based on the instance of origin: some instances are in fact not very active and their contents are lost in the general timeline; an app like Fedilab for example is able to extract the timelines of individual instances, but this function is only allowed by those who use it with Mastodon.

### Describe alternatives you've considered

The alternative is to manually register each user by tracing them to a circle that corresponds to an instance or software

### Additional context

I took the liberty of suggesting this functionality because I noticed, from the administration interface of my instance, that Friendica (unlike Mastodon) already perfectly manages the data relating to the source software.
See here https://INSTANCE/admin/federation for an example

**Comment by annando** [comment]
You can already do so with the help of a user defined channel. We have got a keyword for the used software.

**Comment by informapirata** [comment]
> You can already do so with the help of a user defined channel. We have got a keyword for the used software.

I didn't imagine it could be so easy. Where can I find some more information? Can it also be used to filter instances?

**Comment by annando** [comment]
Yes, you can also filter on the hostname of an instance. See here for more information: https://github.com/friendica/friendica/blob/develop/doc/Channels.md#user-defined-channels

You should test this step by step, since it can be tricky especially when you combine multiple keywords, since you can not only include, but also exclude data. If you are on the develop branch, you can see the search text of individual posts, which can help in finding the best keywords:
![image](https://github.com/friendica/friendica/assets/844208/dd019cbc-1cff-4127-83e0-c2c0d2d59b87)
![image](https://github.com/friendica/friendica/assets/844208/99f9ab03-fe7a-42fd-ad32-d7796f24d5ac)
